### PR TITLE
JSR375:another batch of message update.

### PR DESCRIPTION
**Pull request by toshiyamamoto**

### CODE REVIEW

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 1]
```diff
@@ -19,45 +19,41 @@
...
-JAVAEESEC_WARNING_IDSTORE_CONFIG.explanation=The identity store attribute Expression Language expression cannot be resolved to a valid value.
-JAVAEESEC_WARNING_IDSTORE_CONFIG.useraction=Make sure that the annotation contains a valid configuration value. If you use EL expressions, ensure that they are valid and that any referenced beans that are used in the expression are resolvable.
+JAVAEESEC_WARNING_IDSTORE_CONFIG=CWWKS1916W: The Expression Language (EL) expression for the ''{0}'' attribute of the identity store annotation cannot be resolved to a valid value. Ensure that the EL expression and the result are valid and ensure that any referenced beans that are used in the expression are resolvable. The default attribute value of ''{1}'' is used instead.
+JAVAEESEC_WARNING_IDSTORE_CONFIG.explanation=An error occurred when evaluating the EL expression for the specified attribute. <= Give additional information to explain why it wasn't found, maybe some examples of what the problem could be.
```
Comment (chirp1) [inline dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Remove the "<= Give additonal information....".
Explain what a valid configuration value is or give examples of the most common  values. If you are stuck on what to put here, explain to me what the valid values are and give me some common examples. We can figure out the wording together. Also, we want to start the explanation with something other than "An error occurred..".

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 2]
```diff
@@ -19,45 +19,41 @@
...
 
-JAVAEESEC_WARNING_WRONG_CRED=CWWKS1920W: The credential provided to the IdentityStore is not a UsernamePasswordCredential and cannot be validated.
-JAVAEESEC_WARNING_WRONG_CRED.explanation=The credential provided to the IdentityStore is not a UsernamePasswordCredential and cannot be validated.
-JAVAEESEC_WARNING_WRONG_CRED.useraction=Review the type of credential passed into the IdentityStore.
-
-JAVAEESEC_WARNING_NULL_PWD=CWWKS1921W: A null password was provided to the DatabaseIdentityStore and cannot be validated.
-JAVAEESEC_WARNING_NULL_PWD.explanation=The UsernamePasswordCredential contained a null password.
-JAVAEESEC_WARNING_NULL_PWD.useraction=Supply a valid password.
+JAVAEESEC_ERROR_WRONG_CRED=CWWKS1920E: The credential provided to the IdentityStore is not a UsernamePasswordCredential and cannot be validated.
+JAVAEESEC_ERROR_WRONG_CRED.explanation=The credential provided to the IdentityStore is not a UsernamePasswordCredential and cannot be validated.
```
Comment (chirp1) [inline dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Put a noun after IdentityStore and UsernamePasswordCredential. Doing so provides context for users and translators, i.e. IdentityStore object and UsernamePasswordCredential _blah_

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 3]
```diff
@@ -19,45 +19,41 @@
 # Message prefix block: CWWKS1900 - CWWKS1924
 # 1900 to 1909 will be used for some informational message such as JSR375 is enabled or something like that...
 
-JAVAEESEC_ERROR_NO_IDENTITYSTORES=CWWKS1910E: The IdentityStore configured was not found. Make sure that the IdentityStore configured by the application is available.
-JAVAEESEC_ERROR_NO_IDENTITYSTORES.explanation=The IdentityStore required for authentication was not found.
-JAVAEESEC_ERROR_NO_IDENTITYSTORES.useraction=Investigate any error messages from the IdentityStore.
+JAVAEESEC_ERROR_NO_IDENTITYSTORES=CWWKS1910E: The IdentityStore object configured was not found. Make sure that the IdentityStore object configured by the application is available.
+JAVAEESEC_ERROR_NO_IDENTITYSTORES.explanation=The IdentityStore object required for authentication was not found. <= Give additional information to explain why it wasn't found, maybe some examples of what the problem could be. <<< There are various cause why thie error is reported. A user need to review errors in a message.log file.
```
[the remark this answers is not in the document]
Comment (toshiyamamoto) [reply dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Done

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 4]
```diff
@@ -19,45 +19,41 @@
...
+JAVAEESEC_ERROR_NO_VALIDATION.explanation=No IdentityStore object was found to validate the user. Therefore, the user cannot be authenticated.
+JAVAEESEC_ERROR_NO_VALIDATION.useraction=Make sure that at least one IdentityStore object supports validation of the user.
 
-JAVAEESEC_ERROR_NO_HAM=CWWKS1912E: The HttpAuthenticationMechanism for the {0} module in the {1} application could not be created.
-JAVAEESEC_ERROR_NO_HAM.explanation=The HttpAuthenticationMechanism cannot be created.
-JAVAEESEC_ERROR_NO_HAM.useraction=Investigate any error messages from the HttpAuthenticationMechanism.
+JAVAEESEC_ERROR_NO_HAM=CWWKS1912E: The HttpAuthenticationMechanism object for the {0} module in the {1} application could not be created.
+JAVAEESEC_ERROR_NO_HAM.explanation=The HttpAuthenticationMechanism object cannot be created. <= Give additional information to explain why it wasn't found, maybe some examples of what the problem could be. <<< There are various cause why thie error is reported. A user need to review errors in a message.log file.
```
[the remark this answers is not in the document]
Comment (toshiyamamoto) [reply dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Done.

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 5]
```diff
@@ -19,45 +19,41 @@
...
-JAVAEESEC_WARNING_IDSTORE_CONFIG.useraction=Make sure that the annotation contains a valid configuration value. If you use EL expressions, ensure that they are valid and that any referenced beans that are used in the expression are resolvable.
+JAVAEESEC_WARNING_IDSTORE_CONFIG=CWWKS1916W: The Expression Language (EL) expression for the ''{0}'' attribute of the identity store annotation cannot be resolved to a valid value. Ensure that the EL expression and the result are valid and ensure that any referenced beans that are used in the expression are resolvable. The default attribute value of ''{1}'' is used instead.
+JAVAEESEC_WARNING_IDSTORE_CONFIG.explanation=An error occurred when evaluating the EL expression for the specified attribute. <= Give additional information to explain why it wasn't found, maybe some examples of what the problem could be.
+JAVAEESEC_WARNING_IDSTORE_CONFIG.useraction=Make sure that the annotation contains a valid configuration value. Ensure that the EL expressions are valid and that any referenced beans that are used in the expression are resolvable.
```
[the remark this answers is not in the document]
Comment (toshiyamamoto) [reply dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Some example was added.

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 6]
```diff
@@ -19,45 +19,41 @@
 # Message prefix block: CWWKS1900 - CWWKS1924
 # 1900 to 1909 will be used for some informational message such as JSR375 is enabled or something like that...
 
-JAVAEESEC_ERROR_NO_IDENTITYSTORES=CWWKS1910E: The IdentityStore configured was not found. Make sure that the IdentityStore configured by the application is available.
-JAVAEESEC_ERROR_NO_IDENTITYSTORES.explanation=The IdentityStore required for authentication was not found.
-JAVAEESEC_ERROR_NO_IDENTITYSTORES.useraction=Investigate any error messages from the IdentityStore.
+JAVAEESEC_ERROR_NO_IDENTITYSTORES=CWWKS1910E: The IdentityStore object configured was not found. Make sure that the IdentityStore object configured by the application is available.
+JAVAEESEC_ERROR_NO_IDENTITYSTORES.explanation=The IdentityStore object required for authentication was not found. <= Give additional information to explain why it wasn't found, maybe some examples of what the problem could be. <<< There are various cause why thie error is reported. A user need to review errors in a message.log file.
+JAVAEESEC_ERROR_NO_IDENTITYSTORES.useraction=Investigate any error messages from the IdentityStore object.
```
[the remark this answers is not in the document]
Comment (toshiyamamoto) [reply dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Done.

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 7]
```diff
@@ -19,45 +19,41 @@
...
+JAVAEESEC_ERROR_NO_VALIDATION.explanation=No IdentityStore object was found to validate the user. Therefore, the user cannot be authenticated.
+JAVAEESEC_ERROR_NO_VALIDATION.useraction=Make sure that at least one IdentityStore object supports validation of the user.
 
-JAVAEESEC_ERROR_NO_HAM=CWWKS1912E: The HttpAuthenticationMechanism for the {0} module in the {1} application could not be created.
-JAVAEESEC_ERROR_NO_HAM.explanation=The HttpAuthenticationMechanism cannot be created.
-JAVAEESEC_ERROR_NO_HAM.useraction=Investigate any error messages from the HttpAuthenticationMechanism.
+JAVAEESEC_ERROR_NO_HAM=CWWKS1912E: The HttpAuthenticationMechanism object for the {0} module in the {1} application could not be created.
+JAVAEESEC_ERROR_NO_HAM.explanation=The HttpAuthenticationMechanism object cannot be created. <= Give additional information to explain why it wasn't found, maybe some examples of what the problem could be. <<< There are various cause why thie error is reported. A user need to review errors in a message.log file.
+JAVAEESEC_ERROR_NO_HAM.useraction=Investigate any error messages from the HttpAuthenticationMechanism object.
```
[the remark this answers is not in the document]
Comment (toshiyamamoto) [reply dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Done.

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops (hunk 1 above)
Comment (toshiyamamoto) [reply dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: An example was added.

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops (hunk 2 above)
Comment (toshiyamamoto) [reply dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Done.

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 8]
```diff
@@ -19,45 +19,41 @@
...
-JAVAEESEC_WARNING_IDSTORE_CONFIG.explanation=The identity store attribute Expression Language expression cannot be resolved to a valid value.
-JAVAEESEC_WARNING_IDSTORE_CONFIG.useraction=Make sure that the annotation contains a valid configuration value. If you use EL expressions, ensure that they are valid and that any referenced beans that are used in the expression are resolvable.
+JAVAEESEC_WARNING_IDSTORE_CONFIG=CWWKS1916W: The Expression Language (EL) expression for the ''{0}'' attribute of the identity store annotation cannot be resolved to a valid value. Ensure that the EL expression and the result are valid and ensure that any referenced beans that are used in the expression are resolvable. The default attribute value of ''{1}'' is used instead.
+JAVAEESEC_WARNING_IDSTORE_CONFIG.explanation=The cause of the error is mismatch of the type between the EL result and the expected attribute value. For example, if the expected attribute type is String, the EL result needs to be String.
```
Comment (chirp1) [inline dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Change "is mismatch" to "is a mismatch".

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 9]
```diff
@@ -19,45 +19,41 @@
...
+JAVAEESEC_WARNING_IDSTORE_CONFIG.useraction=Make sure that the annotation contains a valid configuration value. Ensure that the EL expressions are valid and that any referenced beans that are used in the expression are resolvable and the type of the result corresponds with the attribute.
```
Comment (chirp1) [inline dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Change "Ensure that the EL expressions are valid and that any referenced beans that are used in the expression are resolvable and the type of the result corresponds with the attribute." 
to
Ensure that the EL expressions are valid, that any referenced beans that are used in the expression are resolvable, and that the type of the result corresponds with the attribute.

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 10]
```diff
@@ -19,45 +19,41 @@
...
-JAVAEESEC_WARNING_WRONG_CRED=CWWKS1920W: The credential provided to the IdentityStore is not a UsernamePasswordCredential and cannot be validated.
-JAVAEESEC_WARNING_WRONG_CRED.explanation=The credential provided to the IdentityStore is not a UsernamePasswordCredential and cannot be validated.
-JAVAEESEC_WARNING_WRONG_CRED.useraction=Review the type of credential passed into the IdentityStore.
-
-JAVAEESEC_WARNING_NULL_PWD=CWWKS1921W: A null password was provided to the DatabaseIdentityStore and cannot be validated.
-JAVAEESEC_WARNING_NULL_PWD.explanation=The UsernamePasswordCredential contained a null password.
-JAVAEESEC_WARNING_NULL_PWD.useraction=Supply a valid password.
+JAVAEESEC_ERROR_WRONG_CRED=CWWKS1920E: The credential provided to the IdentityStore is not a UsernamePasswordCredential and cannot be validated.
+JAVAEESEC_ERROR_WRONG_CRED.explanation=The credential provided to the IdentityStore object is not a UsernamePasswordCredential class and cannot be validated.
+JAVAEESEC_ERROR_WRONG_CRED.useraction=Review the type of credential passed into the IdentityStore by the HttpAuthMechanism implementation.
```
Comment (chirp1) [inline dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Change to IdentityStore object.

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops (hunk 8 above)
Comment (toshiyamamoto) [reply dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: done.

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops (hunk 9 above)
Comment (toshiyamamoto) [reply dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: done.

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops (hunk 10 above)
Comment (toshiyamamoto) [reply dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: done.

**Comment by chirp1** [comment]
Hi Toshi, the updates look good. I added the "ID reviewed" tag to indicate my approval. - Karen

**Pull request merged**